### PR TITLE
[FIX] l10n_{ca,cn,dk,gt,hn,lt,pt,ve}: remove noupdate=1 for tax template

### PR DESCRIPTION
**Pull request by kitan191**
Tax templates definition of some localizations are set to not update, unlike most of the localizations.
All tax templates should be updatable.

opw-3566568

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by dbkosky — CHANGES_REQUESTED** [review]
I don't think this works as-is (I just tested to make sure).
Just getting rid of `noupdate="1"` and updating the module doesn't change the the value of the `no_update` column in `ir_model_data`. 

Unfortunately it might require something like a migration script & version bump in order to change this for each of these (though I'm not 100% certain).

**Comment by kitan191** [comment]
@dbkosky Thanks for the review.

The fix is not there for existing databases.
It is just to prevent the updating issue for new databases.

For existing databases, a manual operation is needed.

**Comment by dbkosky** [comment]
@kitan191 I agree that the fix is useful for new databases (though it is fixed from saas-16.2 onwards)
In my mind the issue is for people trying to debug any issue that might arise from this change.
After the change it is not easy to tell that the data ever was "noupdate" (one has to navigate through the history of the file to see that it was "noupdate"). 

Maybe I'm exaggerating the problem.  
@Levizar what do you think?

**Comment by Levizar** [comment]
@dbkosky Just for the follow up, we discussed it with @kitan191 and I think you're right.

We've found a solution for it: write migration script at the module level to put the tax.template record of the module to noupdate=False

**Comment by kitan191** [comment]
@dbkosky @Levizar I've updated the fix to add the migration scripts

**Comment by asno-odoo** [comment]
are `account.tax.template` records supposed to be editable by clients?

**Comment by diagnoza** [comment]
> are `account.tax.template` records supposed to be editable by clients?

They are not.

File: addons/l10n_ca/migrations/1.1/pre-migrate.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+def migrate(cr, version):
+    # Set noupdate property of "account.tax.template" records to False
+    cr.execute(
+        """UPDATE ir_model_data
+           SET noupdate=false
+           WHERE module='l10n_ca' AND model='account.tax.template'
+        """
```
Comment (diagnoza) [inline addons/l10n_ca/migrations/1.1/pre-migrate.py:7]: Just a suggestion regarding a cleaner formatting

```suggestion
        """
        UPDATE ir_model_data
           SET noupdate=false
         WHERE module='l10n_ca'
           AND model='account.tax.template'
        """
```

**Comment by dbkosky** [comment]
@asno-odoo 
They shouldn't need to be editable by clients, but they should get updated along with the module.
To clarify, they are used to generate the taxes when loading a chart of accounts (up to saas-16.2, when this whole thing changed).

We sometimes update these tax templates when countries change their tax regimes (VAT rates often change, new rates are introduced etc).

If someone were to install one of these localisations, and then subsequently create another company with one of these charts of accounts (at some later date, after changes to the tax templates have occurred), even with updates to the module applied, they would still find that they have the pre-change taxes, since the tax template records are no-update.

The idea of this is to fix that problem.

File: addons/l10n_ca/migrations/1.1/pre-migrate.py (hunk 1 above)
Comment (kitan191) [reply addons/l10n_ca/migrations/1.1/pre-migrate.py]: Done ;)

**Review by Levizar — APPROVED** [review]
@robodoo r+

**Comment by Levizar** [comment]
@fw-bot up to saas-16.1

**Comment by Levizar** [comment]
@kitan191 Nice job btw!

**Pull request closed without merging**